.docker/Dockerfile

```diff
@@ -26,6 +26,7 @@ RUN apt update && \
     apt install -y \
         build-essential \
         clang-format-14 \
+        clangd-12 \
```

Comment:
Similar to Sebastian's comment: Why do we need this?

Reply:
I added this for linting and LSP support, so that vim has access to the symbols for functionality like Goto definition/declaration etc.